lib/rubygems/bundler_version_finder.rb

```diff
@@ -82,7 +82,15 @@ def self.lockfile_version
   def self.lockfile_contents
     gemfile = ENV["BUNDLE_GEMFILE"]
     gemfile = nil if gemfile && gemfile.empty?
-    Gem::Util.traverse_parents Dir.pwd do |directory|
+
+    pwd = begin
+            Dir.pwd
+          rescue Errno::ENOENT
+            Dir.chdir
+            Dir.pwd
+          end
```

Comment:
I was wondering if there should be a common wrapper for `Dir.pwd`, given its unsafe-ish nature regarding non-existent directories; but I think this should do for now.

Reading up on `Dir.chdir` with no args though, will this work if no `$HOME` env variable is set? It might be safer to not traverse anything in case the current dir doesn't exist (i.e., there's never going to be a Gemfile in the hierarchy, since we can no longer be sure about the hierarchy)

Reply:
Re: `Dir.chdir` good point 🤔

Reply:
The thing is that even if we don't traverse anything, if we don't restore the dir with `Dir.chdir` and `Dir.pwd` gets called in another part of the code it will raise the same error.